openapi-cli/src/main/java/io/ballerina/openapi/cmd/CodeGenerator.java

```diff
@@ -646,17 +646,23 @@ public OpenAPI normalizeOpenAPI(Path openAPIPath, boolean isClient) throws IOExc
      * @throws BallerinaOpenApiException When operationId is missing in any path
      */
     private void validateOperationIds(Set<Map.Entry<String, PathItem>> paths) throws BallerinaOpenApiException {
+        List<String> errorList = new ArrayList<>();
         for (Map.Entry<String, PathItem> entry : paths) {
-            for (Operation operation : entry.getValue().readOperations()) {
-                if (operation.getOperationId() != null) {
-                    String operationId = getValidName(operation.getOperationId(), false);
-                    operation.setOperationId(operationId);
+            for (Map.Entry<PathItem.HttpMethod, Operation> operation :
+                    entry.getValue().readOperationsMap().entrySet()) {
+                if (operation.getValue().getOperationId() != null) {
+                    String operationId = getValidName(operation.getValue().getOperationId(), false);
+                    operation.getValue().setOperationId(operationId);
                 } else {
-                    throw new BallerinaOpenApiException("OperationId is missing for the resource path: "
-                            + entry.getKey());
+                    errorList.add(String.format("OperationId is missing in the resource path: %s(%s)", entry.getKey(),
+                            operation.getKey()));
                 }
             }
         }
+        if (!errorList.isEmpty()) {
+            throw new BallerinaOpenApiException(
+                    "OpenAPI definition has errors: \n\n" + String.join("\n", errorList));
```

Comment:
Adding "\n" in hard code will lead to having some OS dependable issues, so shall we use `System.lineSeparator()` for append a new line?

Reply:
Added